<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<!--....................... CONFIGURABLE PARAMETERS .........................-->

	<env name="GAZEBO_MODEL_PATH" value="$(find bcr_bot)/models:$(optenv GAZEBO_MODEL_PATH)" />
	<env name="GAZEBO_RESOURCE_PATH" value="$(find bcr_bot)/worlds:$(optenv GAZEBO_RESOURCE_PATH)" />
	<arg name="world_name"
		default="$(find aws_robomaker_small_warehouse_world)/worlds/small_warehouse_edit.world" />
	<arg name="gazebo_gui_enabled" default="true" />
	<arg name="two_d_lidar_enabled" default="true" />
	<arg name="camera_enabled" default="true" />
	<arg name="stereo_camera_enabled" default="false" />
	<arg name="wheel_odom_topic" default="odom" />
	<arg name="publish_wheel_odom_tf" default="true" />
	<arg name="conveyor_enabled" default="false" />
	<arg name="ground_truth_frame" default="map" />
	<arg name="odometry_source" default="world" />

	<!-- .......................... INITIAL POSE ................................... -->

	<arg name="bcr_bot_0_initial_pose_x" default="5.0" />
	<arg name="bcr_bot_0_initial_pose_y" default="1.0" />
	<arg name="bcr_bot_0_initial_pose_a" default="0.0" />

	<arg name="bcr_bot_1_initial_pose_x" default="0.0" />
	<arg name="bcr_bot_1_initial_pose_y" default="-8.0" />
	<arg name="bcr_bot_1_initial_pose_a" default="0.0" />

	<arg name="bcr_bot_2_initial_pose_x" default="2.5" />
	<arg name="bcr_bot_2_initial_pose_y" default="8.0" />
	<arg name="bcr_bot_2_initial_pose_a" default="0.0" />

	<arg name="bcr_bot_3_initial_pose_x" default="-2.5" />
	<arg name="bcr_bot_3_initial_pose_y" default="8.0" />
	<arg name="bcr_bot_3_initial_pose_a" default="0.0" />

	<arg name="bcr_bot_4_initial_pose_x" default="-2.5" />
	<arg name="bcr_bot_4_initial_pose_y" default="-8.0" />
	<arg name="bcr_bot_4_initial_pose_a" default="0.0" />

	<!-- ............................... LAUNCH WORLD ................................... -->

	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="world_name" value="$(arg world_name)" />
		<arg name="paused" value="false" />
		<arg name="use_sim_time" value="true" />
		<arg name="gui" value="$(arg gazebo_gui_enabled)" />
		<arg name="headless" value="false" />
		<arg name="debug" value="false" />
	</include>


	<!-- .....................Call the launch file to spawn the robots ................... -->

	<group ns="bcr_bot_0">
		<arg name="robot_namespace" value="bcr_bot_0" />
		<arg name="position_x" value="$(arg bcr_bot_0_initial_pose_x)" />
		<arg name="position_y" value="$(arg bcr_bot_0_initial_pose_y)" />
		<arg name="orientation_yaw" value="$(arg bcr_bot_0_initial_pose_a)" />

		<include file="$(find bcr_bot)/launch/bcr_bot_spawn.launch">
			<arg name="two_d_lidar_enabled" value="$(arg two_d_lidar_enabled)" />
			<arg name="camera_enabled" value="$(arg camera_enabled)" />
			<arg name="stereo_camera_enabled" value="$(arg stereo_camera_enabled)" />
			<arg name="wheel_odom_topic" value="$(arg wheel_odom_topic)" />
			<arg name="publish_wheel_odom_tf" value="$(arg publish_wheel_odom_tf)" />
			<arg name="conveyor_enabled" value="$(arg conveyor_enabled)" />
			<arg name="ground_truth_frame" value="$(arg ground_truth_frame)" />
			<arg name="robot_namespace" value="$(arg robot_namespace)" />
			<arg name="position_x" value="$(arg position_x)" />
			<arg name="position_y" value="$(arg position_y)" />
			<arg name="orientation_yaw" value="$(arg orientation_yaw)" />
			<arg name="odometry_source" value="$(arg odometry_source)" />
		</include>
	</group>

	<group ns="bcr_bot_1">
		<arg name="robot_namespace" value="bcr_bot_1" />
		<arg name="position_x" value="$(arg bcr_bot_1_initial_pose_x)" />
		<arg name="position_y" value="$(arg bcr_bot_1_initial_pose_y)" />
		<arg name="orientation_yaw" value="$(arg bcr_bot_1_initial_pose_a)" />

		<include file="$(find bcr_bot)/launch/bcr_bot_spawn.launch">
			<arg name="two_d_lidar_enabled" value="$(arg two_d_lidar_enabled)" />
			<arg name="camera_enabled" value="$(arg camera_enabled)" />
			<arg name="stereo_camera_enabled" value="$(arg stereo_camera_enabled)" />
			<arg name="wheel_odom_topic" value="$(arg wheel_odom_topic)" />
			<arg name="publish_wheel_odom_tf" value="$(arg publish_wheel_odom_tf)" />
			<arg name="conveyor_enabled" value="$(arg conveyor_enabled)" />
			<arg name="ground_truth_frame" value="$(arg ground_truth_frame)" />
			<arg name="robot_namespace" value="$(arg robot_namespace)" />
			<arg name="position_x" value="$(arg position_x)" />
			<arg name="position_y" value="$(arg position_y)" />
			<arg name="orientation_yaw" value="$(arg orientation_yaw)" />
			<arg name="odometry_source" value="$(arg odometry_source)" />
		</include>
	</group>

	<group ns="bcr_bot_2">
		<arg name="robot_namespace" value="bcr_bot_2" />
		<arg name="position_x" value="$(arg bcr_bot_2_initial_pose_x)" />
		<arg name="position_y" value="$(arg bcr_bot_2_initial_pose_y)" />
		<arg name="orientation_yaw" value="$(arg bcr_bot_2_initial_pose_a)" />

		<include file="$(find bcr_bot)/launch/bcr_bot_spawn.launch">
			<arg name="two_d_lidar_enabled" value="$(arg two_d_lidar_enabled)" />
			<arg name="camera_enabled" value="$(arg camera_enabled)" />
			<arg name="stereo_camera_enabled" value="$(arg stereo_camera_enabled)" />
			<arg name="wheel_odom_topic" value="$(arg wheel_odom_topic)" />
			<arg name="publish_wheel_odom_tf" value="$(arg publish_wheel_odom_tf)" />
			<arg name="conveyor_enabled" value="$(arg conveyor_enabled)" />
			<arg name="ground_truth_frame" value="$(arg ground_truth_frame)" />
			<arg name="robot_namespace" value="$(arg robot_namespace)" />
			<arg name="position_x" value="$(arg position_x)" />
			<arg name="position_y" value="$(arg position_y)" />
			<arg name="orientation_yaw" value="$(arg orientation_yaw)" />
			<arg name="odometry_source" value="$(arg odometry_source)" />
		</include>
	</group>

	<group ns="bcr_bot_3">
		<arg name="robot_namespace" value="bcr_bot_3" />
		<arg name="position_x" value="$(arg bcr_bot_3_initial_pose_x)" />
		<arg name="position_y" value="$(arg bcr_bot_3_initial_pose_y)" />
		<arg name="orientation_yaw" value="$(arg bcr_bot_3_initial_pose_a)" />

		<include file="$(find bcr_bot)/launch/bcr_bot_spawn.launch">
			<arg name="two_d_lidar_enabled" value="$(arg two_d_lidar_enabled)" />
			<arg name="camera_enabled" value="$(arg camera_enabled)" />
			<arg name="stereo_camera_enabled" value="$(arg stereo_camera_enabled)" />
			<arg name="wheel_odom_topic" value="$(arg wheel_odom_topic)" />
			<arg name="publish_wheel_odom_tf" value="$(arg publish_wheel_odom_tf)" />
			<arg name="conveyor_enabled" value="$(arg conveyor_enabled)" />
			<arg name="ground_truth_frame" value="$(arg ground_truth_frame)" />
			<arg name="robot_namespace" value="$(arg robot_namespace)" />
			<arg name="position_x" value="$(arg position_x)" />
			<arg name="position_y" value="$(arg position_y)" />
			<arg name="orientation_yaw" value="$(arg orientation_yaw)" />
			<arg name="odometry_source" value="$(arg odometry_source)" />
		</include>
	</group>

	<group ns="bcr_bot_4">
		<arg name="robot_namespace" value="bcr_bot_4" />
		<arg name="position_x" value="$(arg bcr_bot_4_initial_pose_x)" />
		<arg name="position_y" value="$(arg bcr_bot_4_initial_pose_y)" />
		<arg name="orientation_yaw" value="$(arg bcr_bot_4_initial_pose_a)" />

		<include file="$(find bcr_bot)/launch/bcr_bot_spawn.launch">
			<arg name="two_d_lidar_enabled" value="$(arg two_d_lidar_enabled)" />
			<arg name="camera_enabled" value="$(arg camera_enabled)" />
			<arg name="stereo_camera_enabled" value="$(arg stereo_camera_enabled)" />
			<arg name="wheel_odom_topic" value="$(arg wheel_odom_topic)" />
			<arg name="publish_wheel_odom_tf" value="$(arg publish_wheel_odom_tf)" />
			<arg name="conveyor_enabled" value="$(arg conveyor_enabled)" />
			<arg name="ground_truth_frame" value="$(arg ground_truth_frame)" />
			<arg name="robot_namespace" value="$(arg robot_namespace)" />
			<arg name="position_x" value="$(arg position_x)" />
			<arg name="position_y" value="$(arg position_y)" />
			<arg name="orientation_yaw" value="$(arg orientation_yaw)" />
			<arg name="odometry_source" value="$(arg odometry_source)" />
		</include>
	</group>

</launch>